server/core/src/client_api_whitelist.cpp

```diff
@@ -0,0 +1,239 @@
+#include "client_api_whitelist.hpp"
+
+#include "apiNumber.h"
+#include "rodsErrorTable.h"
+#include "rodsLog.h"
+#include "irods_server_properties.hpp"
+#include "irods_rs_comm_query.hpp"
+
+#include <exception>
+
+namespace
+{
+    auto is_client_agent_connection() -> bool
+    {
+        try {
+            irods::get_server_property<std::string>(irods::AGENT_CONN_KW);
+            return false;
+        }
+        catch (const irods::exception&) {
+            rodsLog(LOG_DEBUG, "Connection is not an agent-to-agent connection");
+        }
+
+        return true;
```

Comment:
i'd prefer something that signals the client-side connection is from a real client app such as icommands.  i feel we are at a point where we need to distinguish the different types of iRODS connection relationships.  that would help with providing a better function name.

however, i actually prefer `client_to_agent_connection` because it is explicit about what conditions need to be satisfied in the `enforce` function.  when i read line 200 with `agent_to_agent_connection`, i feel this loosens the condition in that the reader has to know what the negative space is.

Reply:
i like `is_client_to_agent_connection` too